src/hooks/use-speech-recognition.ts

```diff
@@ -0,0 +1,61 @@
+import { useRef, useState } from 'react';
+import { CapturedSpeech } from '../models/captured-speech';
+import { DeckLanguage, getLanguageCode } from '../models/language';
+
+export function useCaptureSpeech() {
+  const speechRecognitionRef = useRef(getSpeechRecognitionObj());
+  const [isCapturingSpeech, setIsCapturingSpeech] = useState(false);
+
+  return { captureSpeech, stopCapturingSpeech, abortSpeechCapture, isCapturingSpeech };
+
+  function captureSpeech(language: DeckLanguage) {
+    speechRecognitionRef.current.lang = getLanguageCode(language);
```

Comment:
should we have a guard for `speechRecognitionRef.current`?
technically it will be `undefined` for unsupported browsers right?

Reply:
good catch. I fixed the typing on speechRecognitionRef.current and added a guard